File: frontend/src/services/mockData.ts
/** 前端 Mock 数据层.
 *
 * 与后端 mock_data.py 对应，保证前后端数据形状一致。
 * 当 VITE_USE_MOCK_DATA=true 时，api.ts 会走这里而不是 axios。
 */

export interface OverviewItem {
  station_id: number
  name: string
  capacity_kw: number
  daily_energy_kwh: number
  completion_rate: number
  loss_kwh: number
  loss_cny: number
  health_score: number
  pr: number
  status: string
}

function seededRandom(seed: string) {
  let h = 0
  for (let i = 0; i < seed.length; i++) {
    h = (h << 5) - h + seed.charCodeAt(i)
    h |= 0
  }
  return function () {
    h = (h * 9301 + 49297) % 233280
    return h / 233280
  }
}

function todaySeed(prefix: string, id: number | string) {
  const d = new Date().toISOString().slice(0, 10)
  return `${prefix}-${id}-${d}`
}

export const dashboardMock = {
  overview: async () => ({
    station_count: 6,
    online_count: 5,
    offline_count: 1,
    total_capacity_kw: 10500,
    total_active_power_kw: 4820.5,
    total_daily_energy_kwh: 32450.8,
    alarm_summary: { urgent: 2, high: 5, medium: 8, low: 3 },
    system_health: 91.2,
  }),

  stationsOverview: async (): Promise<OverviewItem[]> => {
    const names = ['河西光伏电站', '东山光伏电站', '北区分布式', '临港渔光互补', '沙漠基地一期', '山地光伏二期']
    return names.map((name, idx) => {
      const rng = seededRandom(todaySeed('overview', idx + 1))
      const capacity = 500 + idx * 300
      const completion = 0.75 + rng() * 0.2
      const theoretical = capacity * (3.8 + rng() * 1.2)
      const actual = theoretical * completion
      const loss = Math.max(0, theoretical - actual)
      return {
        station_id: idx + 1,
        name,
        capacity_kw: Math.round(capacity),
        daily_energy_kwh: Math.round(actual),
        completion_rate: Math.round(completion * 100) / 100,
        loss_kwh: Math.round(loss),
        loss_cny: Math.round(loss * 0.42),
        health_score: Math.round(70 + rng() * 28),
        pr: Math.round((0.78 + rng() * 0.14) * 100) / 100,
        status: idx === 5 ? 'inactive' : 'active',
      }
    })
  },

  riskTop: async (limit = 5): Promise<OverviewItem[]> => {
    const all = await dashboardMock.stationsOverview()
    return all.sort((a, b) => a.health_score - b.health_score).slice(0, limit)
  },

  alarmStats: async () => ({ urgent: 2, high: 5, medium: 8, low: 3 }),

  insights: async () => ({
    insight:
      '集团共 6 座场站，2 座健康度低于 80 分，今日预计损失金额 ¥12,480。建议优先处理 [山地光伏二期]（健康度 72 分）。',
  }),
}

export const alarmMock = {
  list: async () => [
    {
      id: 1,
      station_id: 1,
      device_id: 'INV001',
      level: 'critical',
      priority: 'urgent',
      title: '发电异常：辐照充足但无功率',
      description: '逆变器 INV001 辐照 823 W/m²，但功率为 0.12 kW',
      rule_name: 'power_zero_when_sunny',
      status: 'open',
      created_at: new Date(Date.now() - 1000 * 60 * 12).toISOString(),
    },
    {
      id: 2,
      station_id: 2,
      device_id: 'INV003',
      level: 'warning',
      priority: 'high',
      title: 'PR 偏低',
      description: '逆变器 INV003 PR 为 42.3%，低于 50%',
      rule_name: 'low_pr',
      status: 'open',
      created_at: new Date(Date.now() - 1000 * 60 * 35).toISOString(),
    },
    {
      id: 3,
      station_id: 6,
      device_id: 'INV002',
      level: 'critical',
      priority: 'urgent',
      title: '逆变器故障码：1027',
      description: '逆变器 INV002 上报故障码 1027',
      rule_name: 'fault_code',
      status: 'acknowledged',
      created_at: new Date(Date.now() - 1000 * 60 * 58).toISOString(),
    },
    {
      id: 4,
      station_id: 3,
      device_id: 'INV001',
      level: 'warning',
      priority: 'medium',
      title: '通讯中断',
      description: '逆变器 INV001 超过 5 分钟未上报数据',
      rule_name: 'comm_interrupt',
      status: 'open',
      created_at: new Date(Date.now() - 1000 * 60 * 120).toISOString(),
    },
  ],

  summary: async () => [
    { rule_name: 'power_zero_when_sunny', level: 'critical', station_id: 1, count: 2, latest_at: new Date().toISOString() },
    { rule_name: 'low_pr', level: 'warning', station_id: 2, count: 3, latest_at: new Date().toISOString() },
    { rule_name: 'fault_code', level: 'critical', station_id: 6, count: 1, latest_at: new Date().toISOString() },
  ],

  close: async () => ({ success: true }),
  ack: async () => ({ success: true }),
  createWorkOrder: async (alarmId: number) => ({ success: true, data: { work_order_id: alarmId + 100, alarm_id: alarmId, status: 'pending' } }),
}

export const workOrderMock = {
  list: async () => [
    { id: 101, title: '处理告警：发电异常：辐照充足但无功率', priority: 'urgent', status: 'pending', assignee: '张工', station_id: 1, alarm_id: 1, created_at: new Date(Date.now() - 1000 * 60 * 12).toISOString() },
    { id: 102, title: '处理告警：PR 偏低', priority: 'high', status: 'in_progress', assignee: '李工', station_id: 2, alarm_id: 2, created_at: new Date(Date.now() - 1000 * 60 * 35).toISOString() },
    { id: 103, title: '定期清洗组件', priority: 'medium', status: 'completed', assignee: '王工', station_id: 3, alarm_id: null, created_at: new Date(Date.now() - 1000 * 60 * 60 * 24).toISOString() },
  ],

  create: async (data: any) => ({ id: 200, ...data, status: 'pending', created_at: new Date().toISOString() }),

  get: async (id: number) => ({
    id,
    title: '处理告警：发电异常',
    description: '逆变器 INV001 辐照充足但无功率',
    priority: 'urgent',
    status: 'pending',
    assignee: '张工',
    station_id: 1,
    alarm_id: 1,
    feedback: [],
    created_at: new Date().toISOString(),
    updated_at: new Date().toISOString(),
  }),

  updateStatus: async (id: number, status: string, comment?: string, solution?: string) => ({
    id,
    status,
    feedback: [{ status, comment: comment || '', solution: solution || '', created_at: new Date().toISOString() }],
    updated_at: new Date().toISOString(),
  }),

  timeline: async (_id: number) => [
    { status: 'created', comment: '工单创建', created_at: new Date(Date.now() - 1000 * 60 * 60).toISOString() },
    { status: 'in_progress', comment: '已派单处理', created_at: new Date(Date.now() - 1000 * 60 * 30).toISOString() },
    { status: 'completed', comment: '已完成', solution: '更换直流侧保险丝后恢复正常', created_at: new Date().toISOString() },
  ],

  archiveCase: async (id: number) => ({ knowledge_doc_id: id + 1000, title: `案例沉淀：处理告警 ${id}` }),
}

export const metricMock = {
  getStationMetrics: async (id: number) => {
    const rng = seededRandom(todaySeed('metrics', id))
    return {
      station_id: id,
      station_name: `电站 ${id}`,
      timestamp: new Date().toISOString(),
      active_power_kw: Math.round((400 + rng() * 600) * 100) / 100,
      daily_energy_kwh: Math.round((2000 + rng() * 1500) * 100) / 100,
      pr: Math.round((0.78 + rng() * 0.14) * 100) / 100,
      health_score: Math.round(75 + rng() * 23),
    }
  },

  getStationHistory: async (id: number, metric: string) => {
    const rng = seededRandom(`${metric}-${id}`)
    const data: { timestamp: string; value: number }[] = []
    const now = new Date()
    for (let i = 144; i >= 0; i--) {
      const ts = new Date(now.getTime() - i * 10 * 60 * 1000)
      const hour = ts.getHours() + ts.getMinutes() / 60
      let value = 0
      if (metric === 'active_power_kw') {
        if (hour >= 6 && hour <= 18) value = Math.max(0, Math.sin(((hour - 6) / 12) * Math.PI) * 800 + (rng() - 0.5) * 60)
      } else if (metric === 'irradiance_w_m2') {
        if (hour >= 6 && hour <= 18) value = Math.max(0, Math.sin(((hour - 6) / 12) * Math.PI) * 1000 + (rng() - 0.5) * 80)
      } else {
        value = rng() * 100
      }
      data.push({ timestamp: ts.toISOString(), value: Math.round(value * 100) / 100 })
    }
    return data
  },

  getStationsOverview: async () => dashboardMock.stationsOverview(),
  getStationsRanking: async (metric = 'health_score', limit = 10) => {
    const all = await dashboardMock.stationsOverview()
    const reverse = !['loss_cny', 'loss_kwh'].includes(metric)
    return all.sort((a, b) => ((a as any)[metric] - (b as any)[metric]) * (reverse ? -1 : 1)).slice(0, limit)
  },

  getStationEfficiency: async (id: number) => {
    const rng = seededRandom(todaySeed('eff', id))
    const capacity = 1000 + id * 200
    const daily = capacity * (2.5 + rng() * 2)
    const pr = 0.78 + rng() * 0.14
    return {
      station_id: id,
      capacity_kw: capacity,
      daily_energy_kwh: Math.round(daily),
      equivalent_hours: Math.round((daily / capacity) * 100) / 100,
      pr: Math.round(pr * 100) / 100,
      system_efficiency: Math.round(pr * 100 * 100) / 100,
    }
  },

  getStationLosses: async (id: number) => {
    const rng = seededRandom(todaySeed('loss', id))
    const capacity = 1000 + id * 200
    const theoretical = capacity * (3.8 + rng() * 1.2)
    const actual = theoretical * (0.78 + rng() * 0.14)
    const total = Math.max(0, theoretical - actual)
    const breakdown = [
      { name: '辐照损失', kwh: Math.round(total * 0.35), cny: Math.round(total * 0.35 * 0.42) },
      { name: '效率损失', kwh: Math.round(total * 0.3), cny: Math.round(total * 0.3 * 0.42) },
      { name: '故障损失', kwh: Math.round(total * 0.2), cny: Math.round(total * 0.2 * 0.42) },
      { name: '其他损失', kwh: Math.round(total * 0.15), cny: Math.round(total * 0.15 * 0.42) },
    ]
    return {
      station_id: id,
      theoretical_kwh: Math.round(theoretical),
      actual_kwh: Math.round(actual),
      total_loss_kwh: Math.round(total),
      total_loss_cny: Math.round(total * 0.42),
      breakdown,
    }
  },

  getStationHealthTrend: async (id: number, days = 30) => {
    const rng = seededRandom(`health-${id}`)
    const data: { date: string; health_score: number }[] = []
    for (let i = days - 1; i >= 0; i--) {
      const d = new Date(Date.now() - i * 86400000)
      data.push({ date: d.toISOString().slice(0, 10), health_score: Math.round(70 + rng() * 28) })
    }
    return data
  },

  getStationInverters: async (id: number) => {
    const rng = seededRandom(`inv-${id}`)
    return Array.from({ length: 6 }).map((_, i) => {
      const capacity = 100
      const active = capacity * (0.5 + rng() * 0.35)
      return {
        inverter_id: `INV${String(i + 1).padStart(3, '0')}`,
        name: `逆变器 ${i + 1}`,
        capacity_kw: capacity,
        active_power_kw: Math.round(active * 100) / 100,
        daily_energy_kwh: Math.round(active * 3.5 * 100) / 100,
        utilization_rate: Math.round((active / capacity) * 100) / 100,
        status: rng() > 0.85 ? 'offline' : 'online',
      }
    })
  },

  getStationStrings: async (id: number, inverterId?: string) => {
    const rng = seededRandom(`str-${id}-${inverterId || 'all'}`)
    const base = 5.0
    const strings = Array.from({ length: 12 }).map((_, i) => ({
      string_id: `STR${String(i + 1).padStart(3, '0')}`,
      name: `组串 ${i + 1}`,
      inverter_id: inverterId || `INV${String((i % 6) + 1).padStart(3, '0')}`,
      current_a: Math.round(base * (1 + (rng() - 0.5) * 0.16) * 100) / 100,
      capacity_kw: 10,
    }))
    const values = strings.map((s) => s.current_a)
    const avg = values.reduce((a, b) => a + b, 0) / values.length
    const dispersion = (Math.max(...values) - Math.min(...values)) / avg
    return strings.map((s) => ({ ...s, avg_current_a: Math.round(avg * 100) / 100, dispersion_rate: Math.round(dispersion * 100) / 100 }))
  },
}

export const reportMock = {
  list: async () => [],
  generate: async () => ({ id: 1, report_type: 'daily', created_at: new Date().toISOString() }),
  exportPdf: async () => new Blob(['PDF'], { type: 'application/pdf' }),
}

export const diagnosisMock = {
  diagnoseStation: async (id: number) => ({ id: 1, station_id: id, overall_health: 88, summary: '运行总体良好', findings: [], suggestions: [] }),
  listReports: async () => [],
  getReport: async (id: number) => ({ id, station_id: 1, overall_health: 88, summary: '运行总体良好', findings: [], suggestions: [] }),
  exportReportPdf: async () => new Blob(['PDF'], { type: 'application/pdf' }),
  createFeedback: async () => ({ success: true }),
}

export const knowledgeMock = {
  listDocuments: async () => [
    { id: 1, filename: '逆变器故障处理手册.pdf', station_id: null, status: 'active', created_at: new Date().toISOString() },
    { id: 2, filename: '光伏电站运维SOP.txt', station_id: null, status: 'active', created_at: new Date().toISOString() },
  ],
  uploadDocument: async (_file: File, _stationId?: number) => ({
    id: Date.now(),
    filename: _file.name,
    station_id: _stationId || null,
    status: 'active',
    created_at: new Date().toISOString(),
  }),
  deleteDocument: async (_id: number) => ({ success: true }),
  ask: async (question: string, _stationId?: number) => ({
    answer: `这是关于“${question}”的模拟回答。真实环境下会从知识库检索相关内容并返回。`,
    sources: [
      { content: '示例知识片段：逆变器绝缘阻抗低通常由组件或直流线缆受潮、破损引起。', metadata: { filename: '逆变器故障处理手册.pdf' } },
    ],
  }),
}

// 鉴权 / 用户 mock —— 用于无后端的演示环境（GitHub Pages）
export const authMock = {
  login: async ({ username }: { username: string; password: string }) => ({
    access_token: 'mock-token-' + Date.now(),
    user: {
      id: 1,
      username: username || 'demo',
      email: 'demo@pvops.local',
      full_name: '演示用户',
      role: 'admin',
      status: 'active',
    },
  }),
  me: async () => ({
    id: 1,
    username: 'demo',
    email: 'demo@pvops.local',
    full_name: '演示用户',
    role: 'admin',
    status: 'active',
  }),
}

// 电站 / 设备 mock —— 用于演示环境，使无后端的 Pages 也能完整导航
const DEMO_STATIONS = [
  {
    id: 1,
    name: '光伏电站 A',
    code: 'DEMO-001',
    capacity_kw: 1000,
    location: '浙江省杭州市',
    longitude: 120.16,
    latitude: 30.25,
    status: 'active',
    contact_name: '张工',
    contact_phone: '13800138000',
  },
  {
    id: 2,
    name: '光伏电站 B',
    code: 'DEMO-002',
    capacity_kw: 850,
    location: '江苏省苏州市',
    longitude: 120.62,
    latitude: 31.32,
    status: 'active',
    contact_name: '李工',
    contact_phone: '13800138001',
  },
]

const DEMO_DEVICES = (stationId: number) => [
  { id: stationId * 100 + 1, station_id: stationId, device_type: 'weather_station', device_code: 'WS001', name: '气象站 001', protocol: 'simulator', status: 'active', sort_order: 0 },
  { id: stationId * 100 + 2, station_id: stationId, device_type: 'meter', device_code: 'METER001', name: '关口表 001', protocol: 'simulator', status: 'active', sort_order: 1 },
  { id: stationId * 100 + 3, station_id: stationId, device_type: 'inverter', device_code: 'INV001', name: '逆变器 1', vendor: '阳光电源', model: 'SG350HX', protocol: 'simulator', status: 'active', sort_order: 11 },
  { id: stationId * 100 + 4, station_id: stationId, device_type: 'inverter', device_code: 'INV002', name: '逆变器 2', vendor: '阳光电源', model: 'SG350HX', protocol: 'simulator', status: 'active', sort_order: 12 },
  { id: stationId * 100 + 5, station_id: stationId, device_type: 'inverter', device_code: 'INV003', name: '逆变器 3', vendor: '阳光电源', model: 'SG350HX', protocol: 'simulator', status: 'active', sort_order: 13 },
]

export const stationMock = {
  list: async () => DEMO_STATIONS,
  get: async (id: number) => DEMO_STATIONS.find((s) => s.id === id) ?? DEMO_STATIONS[0],
}

export const deviceMock = {
  list: async (stationId?: number) => (stationId ? DEMO_DEVICES(stationId) : DEMO_DEVICES(1)),
  get: async (id: number) => DEMO_DEVICES(Math.floor(id / 100))[0] ?? DEMO_DEVICES(1)[0],
  topology: async (stationId: number) => ({
    station_id: stationId,
    devices: DEMO_DEVICES(stationId).map((d) => ({ ...d, children: [] })),
  }),
}

File: backend/app/services/mock_data.py
"""确定性模拟数据生成器（**已弃用** — 请使用 ``app.demo.MockDataProvider``）.

保留此文件仅为向后兼容：所有业务调用已迁移到 ``app.demo.get_data_provider()``。
新代码不应再 ``from app.services import mock_data``；后续会移除本模块。
"""

import hashlib
import random
from datetime import datetime, timedelta


def _seed(value: str) -> random.Random:
    """基于字符串生成固定随机数生成器."""
    hashed = hashlib.md5(value.encode("utf-8")).hexdigest()
    return random.Random(hashed)


def mock_latest_station_metrics(station_id: int, capacity_kw: float = 1000.0) -> dict:
    """模拟电站最新实时指标."""
    rng = _seed(f"latest-{station_id}-{datetime.now().strftime('%Y%m%d')}")
    # 白天功率曲线
    hour = datetime.now().hour + datetime.now().minute / 60
    if 6 <= hour <= 18:
        peak = ((hour - 6) / 12) * capacity_kw * 0.85
        active_power_kw = max(0, peak + rng.uniform(-capacity_kw * 0.05, capacity_kw * 0.05))
    else:
        active_power_kw = rng.uniform(0, capacity_kw * 0.02)

    daily_energy_kwh = capacity_kw * rng.uniform(2.5, 4.5)
    pr = rng.uniform(0.78, 0.92)
    health_score = rng.uniform(75, 98)

    return {
        "station_id": station_id,
        "station_name": f"电站 {station_id}",
        "timestamp": datetime.now().isoformat(),
        "active_power_kw": round(active_power_kw, 2),
        "daily_energy_kwh": round(daily_energy_kwh, 2),
        "pr": round(pr, 4),
        "health_score": round(health_score, 1),
    }


def mock_metric_history(
    station_id: int,
    metric: str,
    start: datetime | None = None,
    end: datetime | None = None,
    points: int = 144,
) -> list[dict]:
    """模拟指标历史曲线（功率/辐照/温度等）."""
    if end is None:
        end = datetime.now()
    if start is None:
        start = end - timedelta(days=1)

    rng = _seed(f"history-{station_id}-{metric}")
    delta = (end - start) / max(points - 1, 1)
    data = []
    for i in range(points):
        ts = start + delta * i
        hour = ts.hour + ts.minute / 60
        if metric == "active_power_kw":
            if 6 <= hour <= 18:
                peak = ((hour - 6) / 12) * 800
                value = max(0, peak + rng.uniform(-40, 40))
            else:
                value = rng.uniform(0, 10)
        elif metric == "irradiance_w_m2":
            value = max(0, (hour - 6) / 12 * 1000 + rng.uniform(-50, 50)) if 6 <= hour <= 18 else 0
        elif metric == "daily_energy_kwh":
            value = rng.uniform(2000, 3500)
        elif metric in ("dc_voltage_v", "ambient_temp_c"):
            value = rng.uniform(200, 800) if metric == "dc_voltage_v" else rng.uniform(15, 35)
        else:
            value = rng.uniform(0, 100)
        data.append({"timestamp": ts.isoformat(), "value": round(value, 2)})
    return data


def mock_station_overview(stations: list[dict]) -> list[dict]:
    """模拟集团总览数据（用于气泡图/TOP榜）."""
    overview = []
    for station in stations:
        sid = station["id"]
        capacity = station.get("capacity_kw") or 1000
        rng = _seed(f"overview-{sid}")
        theoretical = capacity * rng.uniform(3.8, 5.2)
        actual = theoretical * rng.uniform(0.75, 0.95)
        completion_rate = actual / theoretical if theoretical > 0 else 0
        loss_kwh = max(0, theoretical - actual)
        loss_cny = loss_kwh * 0.42
        health_score = rng.uniform(65, 99)
        pr = rng.uniform(0.75, 0.92)

        overview.append(
            {
                "station_id": sid,
                "name": station.get("name") or f"电站 {sid}",
                "capacity_kw": round(capacity, 2),
                "daily_energy_kwh": round(actual, 2),
                "completion_rate": round(completion_rate, 4),
                "loss_kwh": round(loss_kwh, 2),
                "loss_cny": round(loss_cny, 2),
                "health_score": round(health_score, 1),
                "pr": round(pr, 4),
                "status": station.get("status") or "active",
            }
        )
    return overview


def mock_health_trend(station_id: int, days: int = 30) -> list[dict]:
    """模拟健康度趋势（用于热力图）."""
    rng = _seed(f"health-{station_id}")
    data = []
    end = datetime.now()
    for i in range(days):
        day = end - timedelta(days=days - 1 - i)
        score = rng.uniform(70, 99)
        data.append({"date": day.strftime("%Y-%m-%d"), "health_score": round(score, 1)})
    return data


def mock_inverter_comparison(station_id: int, inverters: list[dict]) -> list[dict]:
    """模拟逆变器群组对比."""
    result = []
    for idx, inv in enumerate(inverters):
        rng = _seed(f"inv-{station_id}-{inv.get('inverter_id', idx)}")
        capacity = inv.get("capacity_kw") or 100
        active_power = capacity * rng.uniform(0.5, 0.85)
        daily_energy = capacity * rng.uniform(2.5, 4.5)
        utilization = active_power / capacity if capacity > 0 else 0
        result.append(
            {
                "inverter_id": inv.get("inverter_id") or f"INV{idx+1:03d}",
                "name": inv.get("name") or f"逆变器 {idx+1}",
                "capacity_kw": round(capacity, 2),
                "active_power_kw": round(active_power, 2),
                "daily_energy_kwh": round(daily_energy, 2),
                "utilization_rate": round(utilization, 4),
                "status": inv.get("status") or "online",
            }
        )
    return result


def mock_string_dispersion(station_id: int, strings: list[dict]) -> list[dict]:
    """模拟组串电流离散."""
    if not strings:
        return []
    rng = _seed(f"string-{station_id}")
    base = 5.0
    data = []
    for idx, s in enumerate(strings):
        variation = rng.uniform(-0.08, 0.08)
        current = base * (1 + variation)
        data.append(
            {
                "string_id": s.get("string_id") or f"STR{idx+1:03d}",
                "name": s.get("name") or f"组串 {idx+1}",
                "inverter_id": s.get("inverter_id") or "INV001",
                "current_a": round(current, 2),
                "capacity_kw": s.get("capacity_kw") or 10,
            }
        )

    values = [float(d.get("current_a", 0) or 0) for d in data]
    avg = sum(values) / len(values)
    dispersion = (max(values) - min(values)) / avg if avg > 0 else 0
    for d in data:
        d["avg_current_a"] = round(avg, 2)
        d["dispersion_rate"] = round(dispersion, 4)
    return data


def mock_efficiency(station_id: int, capacity_kw: float = 1000.0) -> dict:
    """模拟电站效率指标."""
    rng = _seed(f"eff-{station_id}")
    daily_energy = capacity_kw * rng.uniform(2.5, 4.5)
    pr = rng.uniform(0.78, 0.92)
    equivalent_hours = daily_energy / capacity_kw if capacity_kw > 0 else 0
    system_efficiency = pr * 100
    return {
        "station_id": station_id,
        "capacity_kw": round(capacity_kw, 2),
        "daily_energy_kwh": round(daily_energy, 2),
        "equivalent_hours": round(equivalent_hours, 2),
        "pr": round(pr, 4),
        "system_efficiency": round(system_efficiency, 2),
    }


def mock_loss_breakdown(station_id: int, capacity_kw: float = 1000.0) -> dict:
    """模拟损失分解."""
    rng = _seed(f"loss-{station_id}")
    theoretical = capacity_kw * rng.uniform(3.8, 5.2)
    actual = theoretical * rng.uniform(0.78, 0.92)
    total_loss = max(0, theoretical - actual)

    irradiance_loss = total_loss * rng.uniform(0.25, 0.40)
    efficiency_loss = total_loss * rng.uniform(0.20, 0.35)
    fault_loss = total_loss * rng.uniform(0.05, 0.15)
    other_loss = max(0, total_loss - irradiance_loss - efficiency_loss - fault_loss)

    def to_cny(kwh: float) -> float:
        return round(kwh * 0.42, 2)

    return {
        "station_id": station_id,
        "theoretical_kwh": round(theoretical, 2),
        "actual_kwh": round(actual, 2),
        "total_loss_kwh": round(total_loss, 2),
        "total_loss_cny": to_cny(total_loss),
        "breakdown": [
            {"name": "辐照损失", "kwh": round(irradiance_loss, 2), "cny": to_cny(irradiance_loss)},
            {"name": "效率损失", "kwh": round(efficiency_loss, 2), "cny": to_cny(efficiency_loss)},
            {"name": "故障损失", "kwh": round(fault_loss, 2), "cny": to_cny(fault_loss)},
            {"name": "其他损失", "kwh": round(other_loss, 2), "cny": to_cny(other_loss)},
        ],
    }
